unit/ctest_skin_surf_from_ghost.c

```diff
@@ -0,0 +1,398 @@
+/**
+ * This code implements a test suite for validating the `gkyl_skin_surf_from_ghost` 
+ * module, which updates discontinuous Galerkin (DG) coefficients to ensure continuity 
+ * at the interface between skin and ghost cells in structured grids. The update process 
+ * leverages interface-based kernels found in `gkylzero/kernels/skin_surf_from_ghost` 
+ * to adjust skin cell coefficients to match ghost cell values at shared nodes, while 
+ * maintaining the opposing skin cell nodal value.
+ * 
+ * Specifically, the updater enforces that the skin cell value at the interface (e.g., 
+ * upper edge) equals the ghost cell value, achieving continuity across the interface 
+ * while altering the average skin cell value.
+ * 
+ * Example for GKYL_UPPER_EDGE:
+ *   vls            vus  vlg                vug
+ *    |___skin cell___|   |____ghost cell____|
+ *   legend: vls is the lower nodal skin cell value, vus is the upper nodal skin cell value, 
+ *           vlg is the lower nodal ghost cell value, and vug is the upper ghost cell value.
+ *  In this situation, the updater will alter the DG coefficient of the skin cell so that 
+ *    vus = vlg
+ *  keeping vls unchanged.
+ */
+
+#include "gkyl_array.h"
+#include <gkyl_proj_on_basis.h>
+#include <gkyl_range.h>
+#include <gkyl_rect_decomp.h>
+#include <gkyl_rect_grid.h>
+#include <gkyl_array_rio.h>
+#include <acutest.h>
+// This the updater to be tested
+#include <gkyl_skin_surf_from_ghost.h>  
+
+// Evaluate the projection of the modal representation inside a cell from 1x to 3x
+double eval_f(const double *phi, double x, double y, double z, int cdim);
+void compare_skin_ghost_cubes(const double* fskin, const double* fghost, const int cdim, const int edge, const int dir);
+
+// Function to allocate a gkyl array, zero-initialized, on CPU or GPU
+static struct gkyl_array* mkarr(bool on_gpu, long nc, long size) {
+    struct gkyl_array* a;
```

Comment:
@Antoinehoff please edit this file to have a consistent 2-space indentation. In some places it uses 2, in most it uses 4, and in some it even uses 7.